gettype2 prototype implementation

krux02 opened the pull request:
As already mentioned, this gettype2 implementation does not exactly do what it is supposed to do, because I could not find a nice way that would work in all cases without destroying other cases. The idea is that `gettype2` and `getimpl` always return something that resembles the left of the right side of the the type assignment. for types that are builtin the result of `getimpl` is unspecified.

Araq commented:
https://travis-ci.org/nim-lang/Nim/builds/101966218 you even broke booting ;-)

Araq commented on compiler/vmdeps.nim:
```diff
@@ -181,3 +181,65 @@ proc mapTypeToAst(t: PType, info: TLineInfo; allowRecursion=false): PNode =
+proc opMapTypeToAst2(t: PType; info: TLineInfo; getImpl: bool): PNode =
+
+  proc name(tk : TTypeKind): string =
+    if tk == tySequence:
```
Better rename 'tySequence' in the compiler everywhere finally.

krux02 commented:
Yes, I introduced several magics, and added some code duplication, because I do not understand that part good enough to do it without code duplication. But at least, the code duplication is obvious in the code, and next to each other. I hope that helps to get rid of the code duplication easily.

How should I rename `tySequence`?

Araq commented:
Closing since we merged some other PR improving gettype() tremendously.
